requirements/extra.txt

```diff
@@ -0,0 +1,16 @@
+# FORMAT
+# Put your extra requirements here in the following format
+#
+# package[version_required]: tag1, tag2, ...
+#
+# Notes:
+# - Shap is an optional dependency due to build issues, see https://github.com/slundberg/shap/pull/1802
+# - Version choice for shap is related to: https://github.com/SeldonIO/alibi/issues/333
+# - Avoid numba 0.54 due to: https://github.com/SeldonIO/alibi/issues/466
+# - prevent tensorflow 2.6.0 and 2.6.1: https://github.com/SeldonIO/alibi-detect/issues/375
+
+ray>=0.8.7, <2.0.0:                             ray
+shap>=0.40.0, <0.41.0:                          shap
+numba>=0.50.0, !=0.54.0, <0.56.0:               shap
+tensorflow>=2.0.0, !=2.6.0, !=2.6.1, <2.9.0:    tensorflow
```

Comment:
Question: I'm assuming dependabot won't cover these? If not, maybe we need to think about a routine to check for new optional deps releases?

Reply:
hmm, good point. Perhaps something to include in PR templates. Could include a script to automate if possible too

Reply:
We should check first if dependabot definitely ignores this kind of file by e.g. merging this file in a fork.

Reply:
So I tried merging this PR on my own fork and enabling `dependabot`. It seems like the `requirements/extra.txt` file is not tracked by `dependabot`. Frustratingly the documentation on how or what `dependabot` does track is quite vague.

![dependabot-monitoring](https://user-images.githubusercontent.com/29690500/165582340-5af38951-abfa-458c-a410-5ea5f52212d7.png)

Reply:
As agreed elsewhere I've removed the `requirement/extras.txt` reverse index. We're now just using the `extras_require` in `setup.py` as this is tracked by `dependabot`.

Reply:
@mauicv I'm still confused about this, was dependabot actually covering `extras_require`? I can't see any dependabot PR's covering `shap`, `numba` or `ray`? 

https://github.com/SeldonIO/alibi/pulls?page=2&q=is%3Apr+author%3Aapp%2Fdependabot

Reply:
Perhaps worth checking `dependabot` versions and look at their issues/docs to verify? FYI `dependabot` config here: https://github.com/SeldonIO/alibi/blob/master/.github/dependabot.yml

Reply:
I was going off of [this](https://github.com/dependabot/feedback/issues/116) but I'll check

Reply:
Update: looking at the [dependabot logs](https://github.com/SeldonIO/alibi/network/updates/359275579) it does appear `extras_require` is being monitored by dependabot (ctr+f for `ray` etc). Still can't explain the lack of PR's for bumping `ray` etc...